src/_pytest/junitxml.py

```diff
@@ -646,7 +646,7 @@ def pytest_sessionstart(self) -> None:
     def pytest_sessionfinish(self) -> None:
         dirname = os.path.dirname(os.path.abspath(self.logfile))
         if not os.path.isdir(dirname):
-            os.makedirs(dirname)
+            os.makedirs(dirname, exist_ok=True)
```

Comment:
Probably makes sense to also remove the `if` condition. Also probably nice to leave a comment in the code explaining the rationale around that.

Other than that, can you please also add a changelog entry as "bugfix"?